DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,283 @@
+- [`Node.js`](https://nodejs.org/), [`npm`](https://www.npmjs.com/), and [`yarn`](https://yarnpkg.com/) for building and running OpenSearch Dashboards
+- A code editor like [Visual Studio Code](https://code.visualstudio.com/) configured for JavaScript/TypeScript development
+
+
+### Install `git`
+
+Check if Git is installed:
+
+```bash
+git --version
+```
+#### Install `git`
+
+If you don't already have it installed (check with `git --version`) we recommend following the [the `git` installation guide for your OS](https://git-scm.com/downloads).
+
+#### Install `node`
+
+We recommend using [Node Version Manager (nvm)](https://github.com/nvm-sh/nvm) to install and manage different node versions, which may differ between release branches.
+
+1. Install nvm (as specified by the [`nvm` README](https://github.com/nvm-sh/nvm#installing-and-updating)): `curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.3/install.sh | bash`
+2. Install the version of the Node.js runtime defined in [`.nvmrc`](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/.nvmrc): `nvm install`
+
+If it's the only version of node installed, it will automatically be set to the `default` alias. Otherwise, use `nvm list` to see all installed `node` versions, and `nvm use` to select the node version required by OpenSearch Dashboards.
+
+## Fork and Clone Repositories
+
+You will be working with two repositories:
+
+1. **OpenSearch Dashboards** — the core platform where the plugin runs.
+2. **Query Insights Dashboards Plugin** — the plugin you are developing.
+
+---
+
+### Step 1: Fork the Repositories
+
+Fork both of the following repositories to your own GitHub account:
+
+- [OpenSearch Dashboards](https://github.com/opensearch-project/OpenSearch-Dashboards)
+- [Query Insights Dashboards Plugin](https://github.com/opensearch-project/query-insights-dashboards)
+
+
+### Step 2: Clone OpenSearch Dashboards
+
+Clone your fork of **OpenSearch Dashboards** to your local machine:
+
+```bash
+git clone https://github.com/<your-username>/OpenSearch-Dashboards.git
+cd OpenSearch-Dashboards
+```
+
+
+### Step 3: Clone the Plugin into the plugins/ Directory
+Inside the OpenSearch-Dashboards directory:
+
+```
+cd plugins
+git clone https://github.com/<your-username>/query-insights-dashboards.git
+cd ..
+```
+
+### File Structure
+
+The file structure should be this
+
+```
+OpenSearch-Dashboards/
+├── plugins/
+    └── query-insights-dashboards/
+```
+
+
+
+##  Bootstrap OpenSearch Dashboards with the Query Insights Plugin
+
+After cloning both the OpenSearch Dashboards repository and placing the Query Insights plugin inside the `plugins/` folder, you need to bootstrap the environment. This installs all dependencies and registers the plugin with OpenSearch Dashboards.
+
+---
+
+### Step 1: Install Dependencies and Build
+
+From the root of your `OpenSearch-Dashboards\plugins` directory, run:
+
+```bash
+cd..
+yarn osd bootstrap
```

Comment:
`This command installs dependencies for both the Dashboards core and all plugins, including Query Insights.`